examples/platform/silabs/efr32/rs911x/hal/sl_si91x_ncp_utility.c

```diff
@@ -0,0 +1,282 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * Includes
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "FreeRTOS.h"
+#include "event_groups.h"
+#include "semphr.h"
+#include "task.h"
+#include "dmadrv.h"
+#include "em_chip.h"
+#include "em_cmu.h"
+#include "em_core.h"
+#include "em_device.h"
+#include "em_gpio.h"
+#include "em_ldma.h"
+#include "gpiointerrupt.h"
+#include "spidrv.h"
+
+#include "sl_device_init_clocks.h"
+#include "sl_device_init_hfxo.h"
+#include "sl_status.h"
+
+#include "silabs_utils.h"
+#include "sl_si91x_ncp_utility.h"
+#include "wfx_host_events.h"
+#include "wfx_rsi.h"
+
+#if SL_MX25CTRL_MUX
+sl_status_t sl_wfx_host_spiflash_cs_assert(void);
+sl_status_t sl_wfx_host_spiflash_cs_deassert(void);
+#endif
+
+#if SL_BTLCTRL_MUX
+#include "btl_interface.h"
+#endif // SL_BTLCTRL_MUX
+#if SL_LCDCTRL_MUX
+#include "sl_memlcd.h"
+#endif // SL_LCDCTRL_MUX
+#if SL_MX25CTRL_MUX
+#include "sl_mx25_flash_shutdown_usart_config.h"
+#endif // SL_MX25CTRL_MUX
+
+#define CONCAT(A, B) (A##B)
+#define SPI_CLOCK(N) CONCAT(cmuClock_USART, N)
+
+#if SL_SPICTRL_MUX
+StaticSemaphore_t spi_sem_peripheral;
+SemaphoreHandle_t spi_sem_sync_hdl;
+#endif // SL_SPICTRL_MUX
+
+#if SL_LCDCTRL_MUX
+
+/********************************************************
+ * @fn   sl_wfx_host_pre_lcd_spi_transfer(void)
+ * @brief
+ *        Deal with the SPI PINS MUX that are associated with LCD. for setup before data transfer PINS.
+ * @return
+ *        None
+ **********************************************************/
+sl_status_t sl_wfx_host_pre_lcd_spi_transfer(void)

```

Comment:
isn't the majority of this code already in efx_spi.c?

Reply:
Yeah we will create this so both efx_spi as well as efx32_ncp_host files can you have this utility function going further soon we will remove it from efx_spi file and use this file for definitions